src/App.tsx

```diff
@@ -1,61 +1,114 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './interfaces/todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId);
+};
+
+const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const id = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => ([
+      ...currentTodos,
+      newTodo,
+    ]));
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setErrorTitle(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setErrorUser(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    setErrorTitle(!trimmedTitle);
+    setErrorUser(userId === 0);
+
+    if (trimmedTitle && userId > 0) {
+      addTodo({
+        id,
+        title,
+        completed: false,
+        userId,
+        user: getUser(userId),
+      });
+
+      setTitle('');
+      setUserId(0);
+    }
```

Comment:
let's improve your code 
```suggestion
    if (!trimmedTitle ||!userId 0) {
     return;
    }
    addTodo({
      id,
      title,
      completed: false,
      userId,
      user: getUser(userId),
     });

     setTitle('');
     setUserId(0);
```